enos/README.md

```diff
@@ -0,0 +1,113 @@
+# Upgrade Testing with Enos
+
+We're using [Enos](https://github.com/hashicorp/enos) to perform upgrade
+testing. These tests are run via GitHub Actions from the private `nomad-e2e`
+repository. This document describes how you can run these tests from your local
+development environment if you're a HashiCorp developer.
+
+There are two major components to be aware of:
+* This directory includes the upgrade scenario and the Terraform modules and
+  shell scripts needed to execute that scenario.
+* The scenario uses the same cluster provisioning infrastructure as the E2E
+  tests in the `e2e/` directory in the root of this repo. So to run the upgrade
+  scenario you also have to have all the credentials set up to run the E2E
+  tests. (We may try to fold these together in the future.)
+
+The `terraform/` folder has provisioning code to spin up a Nomad cluster on
+AWS. You'll need both Terraform and AWS credentials to setup AWS instances on
+which e2e tests will run. See the
+[README](https://github.com/hashicorp/nomad/blob/main/e2e/terraform/README.md)
+for details. The number of servers and clients is configurable, as is the
+specific build of Nomad to deploy and the configuration file for each client
+and server.
+
+## Setup
+
+You'll need a recent version of Terraform, the most current version of Enos, as
+well as AWS credentials to create the Nomad cluster and credentials for HCP. The
+Terraform configurations assume that an appropriate instance role has been
+configured elsewhere and that you have the ability to `AssumeRole` into the AWS
+account.
+
+Configure the following environment variables. For HashiCorp Nomad developers,
+this configuration can be found in 1Pass in the Nomad team's vault under
+`nomad-e2e`.
+
+```
+export HCP_CLIENT_ID=
+export HCP_CLIENT_SECRET=
+```
+
+The Vault admin token will expire after 6 hours. If you haven't created one
+already use the separate Terraform configuration found in the
+`$REPO/e2e/terraform/hcp-vault-auth` directory. The following will set the correct
+values for `VAULT_TOKEN`, `VAULT_ADDR`, and `VAULT_NAMESPACE`:
+
+```
+terraform init
+terraform apply --auto-approve
+$(terraform output --raw environment)
+```
+
+Make sure your AWS credentials have been refreshed with the appropriate IAM role:
+
+```
+$ doormat login --force
+$ doormat aws cred-file add-profile --role "$ROLE" --set-default
+```
+
+Next you'll need to obtain an Artifactory token via Doormat.
+
+```
+export ARTIFACTORY_TOKEN=$(doormat artifactory create-token | jq -r .access_token)
+```
+
+Next you'll need to create an Enos variables file (unlike Terraform, Enos
+doesn't accept variables on the command line). Create a file with the following
+values:
```

Comment:
If you create a file `enos.vars.hcl` under the same directory, they are picked automagically by enos